### PR TITLE
Setup CI for main branch

### PR DESCRIPTION
**Pull request by Everest2020**
In order to keep the project healthy, we would like to setup CI.
- Use dotnet CLI to build the solution.
- Use dotnet CLI to run the tests.

### CODE REVIEW

**Review by SukruthGrandhiNI — APPROVED** [review]

**Review by jryckman — COMMENTED** [review]
Is there a way to add a stage to this pipeline that runs all the test in the test assemblies?

**Comment by Everest2020** [comment]
> Is there a way to add a stage to this pipeline that runs all the test in the test assemblies?

I already added one, 

>   - name: Run Tests
        run: dotnet test NationalInstruments.Analyzers.sln

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: windows-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+        name: Checkout code
```
Comment (jryckman) [inline .github/workflows/ci.yml]: I don't fully understand yml, but it seems weird that the first usage here has `-uses`  first with the dash in front of it and name below, where everything else is the opposite. Is that intentional?

Comment (Everest2020) [reply .github/workflows/ci.yml]: The order does not matter. According to the yaml syntax, the first section should begin with '-'

**Review by jryckman — APPROVED** [review]

**Pull request merged**